基于接口的自定义枚举类转换失败

**Describe the bug**
几个关键的java类定义如下
```
@Data
public class UserSave {
    /**
     * 性别
     */
    private Gender sex;
}

public enum Gender implements IEnumWithLabel {
    male("1", "男"),
    female("2", "女"),
    other("0", "未知");

    private String value;
    private String label;

    Gender(String value, String label) {
        this.value = value;
        this.label = label;
    }


    @Override
    public String getValue() {
        return value;
    }

    @Override
    public String getLabel() {
        return label;
    }
}

@RestController
@RequestMapping("/user")
public class UserController extends BaseRestController {
    /**
     * 新增用户信息
     * @param userSave
     * @return
     */
    @PostMapping("/saveUser")
    public Id<Integer> saveUser(@RequestBody UserSave userSave) {
        return Id.build(userService.saveUser(userSave));
    }
}
```
其中.yapi.config 根据官方文档定义：
```
json.rule.enum.convert[groovy:it.extend("com.common.base.IEnumWithLabel")]=~#value
```
报错异常：
```
[ERROR]	error eval script:it.extend("com.common.base.IEnumWithLabel")
[TRACE]	javax.script.ScriptException: javax.script.ScriptException: groovy.lang.MissingMethodException: No signature of method: com.itangcent.idea.plugin.rule.ScriptRuleParser$ScriptDuckTypeContext.extend() is applicable for argument types: (String) values: [com.common.base.IEnumWithLabel]
Possible solutions: isExtend(java.lang.String), exts(), every(), find(), getExt(java.lang.String), getAt(java.lang.String)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:158)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser.eval(ScriptRuleParser.kt:89)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser.access$eval(ScriptRuleParser.kt:34)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser$parseBooleanRule$1.invoke(ScriptRuleParser.kt:62)
	at com.itangcent.idea.plugin.rule.ScriptRuleParser$parseBooleanRule$1.invoke(ScriptRuleParser.kt:34)
	at com.itangcent.intellij.config.rule.BooleanRule$Companion$of$2.compute(BooleanRule.kt:26)
	at com.itangcent.intellij.config.rule.BooleanRule$Companion$of$2.compute(BooleanRule.kt:24)
	at com.itangcent.intellij.config.rule.StringRule$Companion$filterWith$1.invoke(StringRule.kt:15)
	at com.itangcent.intellij.config.rule.StringRule$Companion$filterWith$1.invoke(StringRule.kt:4)
	at com.itangcent.intellij.config.rule.StringRule$Companion$of$1.compute(StringRule.kt:8)
	at com.itangcent.intellij.config.rule.StringRule$Companion$of$1.compute(StringRule.kt:6)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$compute$1.apply(DefaultRuleComputer.kt:64)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer$compute$1.apply(DefaultRuleComputer.kt:9)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1632)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.compute(DefaultRuleComputer.kt:66)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.computer(DefaultRuleComputer.kt:30)
	at com.itangcent.intellij.config.rule.RuleComputer$DefaultImpls.computer(RuleComputer.kt:16)
	at com.itangcent.intellij.config.rule.DefaultRuleComputer.computer(DefaultRuleComputer.kt:9)
	at com.itangcent.intellij.psi.DefaultPsiClassHelper.parseFieldOrMethod(DefaultPsiClassHelper.kt:145)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper$getFields$2.invoke(AbstractPsiClassHelper.kt:450)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper$getFields$2.invoke(AbstractPsiClassHelper.kt:32)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.foreachField(AbstractPsiClassHelper.kt:490)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.getFields(AbstractPsiClassHelper.kt:434)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.getTypeObject(AbstractPsiClassHelper.kt:410)
	at com.itangcent.intellij.psi.AbstractPsiClassHelper.getTypeObject(AbstractPsiClassHelper.kt:296)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter.processMethodParameters(AbstractRequestClassExporter.kt:478)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter.exportMethodApi(AbstractRequestClassExporter.kt:170)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter.access$exportMethodApi(AbstractRequestClassExporter.kt:36)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter$export$1.invoke(AbstractRequestClassExporter.kt:127)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter$export$1.invoke(AbstractRequestClassExporter.kt:36)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter$sam$java_util_function_Consumer$0.accept(AbstractRequestClassExporter.kt)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter.foreachMethod(AbstractRequestClassExporter.kt:453)
	at com.itangcent.idea.plugin.api.export.AbstractRequestClassExporter.export(AbstractRequestClassExporter.kt:123)
	at com.itangcent.idea.plugin.api.export.ComboClassExporter$export$1.apply(ComboClassExporter.kt:41)
	at com.itangcent.idea.plugin.api.export.ComboClassExporter$export$1.apply(ComboClassExporter.kt:15)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)
	at com.itangcent.idea.plugin.api.export.ComboClassExporter.export(ComboClassExporter.kt:41)
	at com.itangcent.idea.plugin.api.export.yapi.YapiApiExporter$doExport$3.invoke(YapiApiExporter.kt:68)
	at com.itangcent.idea.plugin.api.export.yapi.YapiApiExporter$doExport$3.invoke(YapiApiExporter.kt:17)
	at com.itangcent.intellij.psi.SelectedHelper$Builder$onFile$1.invoke(SelectedHelper.kt:151)
	at com.itangcent.intellij.psi.SelectedHelper$Builder$onFile$1.invoke(SelectedHelper.kt:23)
	at com.itangcent.intellij.context.ActionContext$runInReadUI$1.run(ActionContext.kt:324)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:52)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:865)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:51)
	at com.itangcent.intellij.context.ActionContext.runInReadUI(ActionContext.kt:318)
	at com.itangcent.intellij.psi.SelectedHelper$Builder.onFile(SelectedHelper.kt:149)
	at com.itangcent.intellij.psi.SelectedHelper$Builder.access$onFile(SelectedHelper.kt:23)
	at com.itangcent.intellij.psi.SelectedHelper$Builder$traversal$2.invoke(SelectedHelper.kt:84)
	at com.itangcent.intellij.psi.SelectedHelper$Builder$traversal$2.invoke(SelectedHelper.kt:23)
	at com.itangcent.intellij.context.ActionContext$runInReadUI$1.run(ActionContext.kt:324)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:52)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:869)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:51)
	at com.itangcent.intellij.context.ActionContext.runInReadUI(ActionContext.kt:318)
	at com.itangcent.intellij.psi.SelectedHelper$Builder.traversal(SelectedHelper.kt:78)
	at com.itangcent.idea.plugin.api.export.yapi.YapiApiExporter.doExport(YapiApiExporter.kt:76)
	at com.itangcent.idea.plugin.api.export.yapi.YapiApiExporter.export(YapiApiExporter.kt:22)
	at com.itangcent.idea.plugin.actions.YapiExportAction.actionPerformed(YapiExportAction.kt:45)
	at com.itangcent.intellij.actions.KotlinAnAction$actionPerformed$2.invoke(KotlinAnAction.kt:49)
	at com.itangcent.intellij.actions.KotlinAnAction$actionPerformed$2.invoke(KotlinAnAction.kt:18)
	at com.itangcent.intellij.context.ActionContext$runAsync$2.run(ActionContext.kt:173)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: javax.script.ScriptException: groovy.lang.MissingMethodException: No signature of method: com.itangcent.idea.plugin.rule.ScriptRuleParser$ScriptDuckTypeContext.extend() is applicable for argument types: (String) values: [com.common.base.IEnumWithLabel]
Possible solutions: isExtend(java.lang.String), exts(), every(), find(), getExt(java.lang.String), getAt(java.lang.String)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:320)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:155)
	... 97 more
Caused by: groovy.lang.MissingMethodException: No signature of method: com.itangcent.idea.plugin.rule.ScriptRuleParser$ScriptDuckTypeContext.extend() is applicable for argument types: (String) values: [com.common.base.IEnumWithLabel]
Possible solutions: isExtend(java.lang.String), exts(), every(), find(), getExt(java.lang.String), getAt(java.lang.String)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:70)
	at org.codehaus.groovy.runtime.callsite.PojoMetaClassSite.call(PojoMetaClassSite.java:46)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)
	at Script774.run(Script774.groovy:1)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:317)
	... 98 more
```

将表示改为
```
json.rule.enum.convert[groovy:it.isExtend("com.common.base.IEnumWithLabel")]=~#value
```
不报错但是生成的yapi文档没达到预想的效果。

--- a/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/rule/ScriptRuleParser.kt
+++ b/idea-plugin/src/main/kotlin/com/itangcent/idea/plugin/rule/ScriptRuleParser.kt
@@ -309,14 +309,7 @@ abstract class ScriptRuleParser : RuleParser {
         }
 
         override fun isExtend(superClass: String): Boolean {
-            var currClass: PsiClass? = psiClass
-            do {
-                if (superClass == currClass!!.qualifiedName) {
-                    return true
-                }
-                currClass = currClass.superClass
-            } while (currClass != null && currClass.name != "Object")
-            return false
+            return jvmClassHelper!!.isInheritor(psiClass, superClass)
         }
 
         override fun isMap(): Boolean {
@@ -744,16 +737,7 @@ abstract class ScriptRuleParser : RuleParser {
         }
 
         override fun isExtend(superClass: String): Boolean {
-            return getResource()?.let { psiClass ->
-                var currClass: PsiClass? = psiClass as PsiClass
-                do {
-                    if (superClass == currClass!!.qualifiedName) {
-                        return true
-                    }
-                    currClass = currClass.superClass
-                } while (currClass != null && currClass.name != "Object")
-                return false
-            } ?: false
+            return jvmClassHelper!!.isInheritor(psiType, superClass)
         }
 
         override fun isMap(): Boolean {
@@ -876,16 +860,7 @@ abstract class ScriptRuleParser : RuleParser {
         }
 
         override fun isExtend(superClass: String): Boolean {
-            return getResource()?.let { psiClass ->
-                var currClass: PsiClass? = psiClass as PsiClass
-                do {
-                    if (superClass == currClass!!.qualifiedName) {
-                        return true
-                    }
-                    currClass = currClass.superClass
-                } while (currClass != null && currClass.name != "Object")
-                return false
-            } ?: false
+            return jvmClassHelper!!.isInheritor(duckType, superClass)
         }
 
         override fun isMap(): Boolean {

